File: Graphs/dfs.java

import java.util.*;

public class Main {

    static int n,m;
    static List<Integer> [] g;
    static boolean [] vis;
    
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        n = in.nextInt();
        m = in.nextInt();
        
        g = new List[n];
        for(int i=0; i<n; i++)
            g[i] = new ArrayList();
        
        for(int i=0; i<m; i++) {
            int a = in.nextInt();
            int b = in.nextInt();
            
            g[a].add(b); // a -> b
            g[b].add(a); // b -> a  (for un-directed graph)
        }
        
        vis = new boolean[n];
        dfs(0); // or    dfs_stack(0);
        
    }
    
    static void dfs(int u) {
        vis[u] = true;
        
        for(int w : g[u]) 
            if(!vis[w]) 
                dfs(w);
    }
    
    static void dfs_stack(int u) {
        Stack<Integer> s = new Stack();
        
        vis[u] = true;
        s.add(u);
        
        while(!s.isEmpty()) {
            int v = s.pop();
            
            for(int w : g[v]) {
                if(!vis[w]) {
                    vis[w] = true;
                    s.add(w);
                }
            }
        }
    }
}

File: Graphs/dfs.cpp

// ~/BAU/ACM-ICPC/Teams/A++/BlackBurn95
// ~/sudo apt-get Accpeted

#include <iostream>
#include <stdio.h>
#include <memory.h>
#include <algorithm>
#include <vector>
#include <stack>


using namespace std;

int n, m;
vector<vector<int> > g;
bool vis[10001];

void dfs(int u) {
	vis[u] = 1;

	for (int i = 0; i < g[u].size(); i++)
		if (!vis[g[u][i]])
			dfs(g[u][i]);
}

void dfs_stack(int u) {
	stack<int> s;
	vis[u] = 1;
	s.push(u);

	while (!s.empty()) {
		int v = s.top();
		s.pop();

		for (int i = 0; i < g[v].size(); i++)
			if (!vis[g[v][i]]) {
				vis[g[v][i]] = 1;
				s.push(g[v][i]);
			}
	}
}

int main() {
	scanf("%d%d", &n, &m);

	g.clear();
	g.resize(n);

	for (int i = 0; i < m; i++) {
		int a, b;
		scanf("%d%d", &a, &b);

		g[a].push_back(b); // a -> b
		g[b].push_back(a); // b -> a (for un-directed graph)
	}

	memset(vis, 0, sizeof(vis));

	dfs(0); // or    dfs_stack(0);

	return 0; 
}